### PR TITLE
Mes 2073 test report performance

### PR DESCRIPTION
**Pull request by buzzub**
## Description and relevant Jira numbers

Bit of a big PR...

Update the components on the Test Report page so that they are more responsive and don't feel sluggish. Demo on my iPad is better than a screenshot :) 

- Override the default hammer press time so that it's 301ms (instead of 251ms)
- Change ETA and Legal Requirements components into smaller, reusable ones
- Change ETA and Legal Requirement actions to generic, accepting a payload instead
- Use tappable div in place of buttons to improve performance
- Update tests
- Inherit styles for test report buttons from shared CSS
- Manually add and remove an activated class when the buttons are tapped using onTouchStart and onTouchEnd events.
- Delay the removal of the activated class by 100ms. This is necessary when visible feedback is needed for a short tap as the page needs time to render the class before removing it.

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

IMG_FFE934677F7F-1

### CODE REVIEW

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts [hunk 1]
```diff
@@ -2,9 +2,7 @@ import { ComponentFixture, async, TestBed } from '@angular/core/testing';
 import { CompetencyComponent } from '../competency';
 import { AppModule } from '../../../../../app/app.module';
 import { By } from '@angular/platform-browser';
-import { HammerProvider } from '../../../../../providers/hammer/hammer';
-import { MockElementRef } from '../../../../../shared/mocks/element-ref.mock';
-import { Renderer2 } from '@angular/core';
+// import { MockElementRef } from '../../../../../shared/mocks/element-ref.mock';
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/competency/__tests__/competency.spec.ts]: remove comment

Comment (buzzub) [reply src/pages/test-report/components/competency/__tests__/competency.spec.ts]: Done.

File: src/pages/test-report/components/competency/competency.ts [hunk 2]
```diff
@@ -46,15 +41,19 @@ interface CompetencyState {
...
   competency: Competencies;
 
-  @ViewChild('competencyButton')
-  button: ElementRef;
+  touchStateDelay: number = 100;
+
+  touchState: boolean = false;
+  rippleState: boolean = false;
+
+  rippleTimeout: any;
```
Comment (ammarhaiderbjss) [inline src/pages/test-report/components/competency/competency.ts]: Apparently `setTimeout` returns a positive integer value as the timeoutId - https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Return_value - So wonder if setting the type as `number` would work

Comment (buzzub) [reply src/pages/test-report/components/competency/competency.ts]: Setting it as number gives a TypeScript error :(

**Pull request merged**